torch/csrc/distributed/c10d/Backend.hpp

```diff
@@ -17,7 +17,7 @@
 #include <torch/csrc/distributed/c10d/debug.h>
 
 constexpr auto kBackendDefaultTimeout =
-    std::chrono::milliseconds(30 * 60 * 1000);
```

Comment:
Actually I am thinking of whether we can make this configurable by users? I mean 10 mins sound make sense to me but what if users want it to stay alive longer so that they can gdb into the host and debug?

Reply:
oh, it is already configurable via `dist.init_process_group(, timeout=<timeout>)`

But there is no env, which is a possible gap we could address.  I'm not sure if in practice trainers expose an ENV.

Should we just make it an ENV now?

Reply:
Hmm Got it, then it's trainer's responsibility to specify this. I didn't know it is already configurable.

Reply:
just wondering - will it be "safer" if we make the 10min change just for nccl, to avoid breaking things like cpu training which uses gloo?

Reply:
Yea, if you think cpu training needs 30min timeout then I can change just the nccl timeout.